src/incidents/incident-slice.ts

```diff
@@ -120,4 +124,23 @@ export const reportIncident = (
   }
 }
 
+export const completeIncident = (
+  incidentToComplete: Incident,
+  onSuccess?: (incidentToComplete: Incident) => void,
+): AppThunk => async (dispatch) => {
+  dispatch(completeIncidentStart())
+
+  const completedIncident = await IncidentRepository.saveOrUpdate({
+    ...incidentToComplete,
+    completedOn: new Date(Date.now().valueOf()).toISOString(),
+    status: 'completed',
+  })
+
+  dispatch(completeIncidentSuccess(completedIncident))
+
+  if (onSuccess) {
+    onSuccess(completedIncident)
+  }
+}
```

Comment:
there should be a test for this functionality

Reply:
I have added an 'on resolve' test into the ViewIncident test file, will that be ok?

Reply:
Actually, just had a look at some lab module examples, I'll go ahead write tests in the incident-slice as well.